viennacl/src/main/scala/org/apache/mahout/viennacl/opencl/javacpp/DenseColumnMatrix.scala

```diff
@@ -36,12 +36,24 @@ import org.bytedeco.javacpp.annotation._
 @Name(Array("viennacl::matrix<double,viennacl::column_major>"))
 final class DenseColumnMatrix(initDefault:Boolean = true) extends MatrixBase {
 
-  def this(nrow: Int, ncol: Int, ctx: Context = new Context()) {
+  def this(nrow: Int, ncol: Int) {
+    this(false)
+    allocate(nrow, ncol, new Context())
+  }
+
+  def this(nrow: Int, ncol: Int, ctx: Context) {
     this(false)
     allocate(nrow, ncol, ctx)
   }
 
-  def this(data: DoublePointer, nrow: Int, ncol: Int, ctx: Context = new Context(Context.MAIN_MEMORY)) {
```

Comment:
@rawkintrevo lgtm - ship it! we swill still need to do some testing against other spark versions though.  As well, there are some other issues that I've noticed, but can be addressed in other PRs on top of this anyways, eg. #327.